mulog-prometheus/src/com/brunobonacci/mulog/publishers/prometheus/metrics_spec.clj

```diff
@@ -0,0 +1,37 @@
+(ns com.brunobonacci.mulog.publishers.prometheus.metrics_spec
+  (:require [clojure.spec.alpha :as s]))
+
+(def metric-name-re                #"[a-zA-Z0-9_:]*")
+(def metric-label-name-re          #"[a-zA-Z0-9_]*")
+(def reserved-metric-label-name-re #"__.*")
+(def sanitize-metric-name-re       #"[^a-zA-Z0-9_:]+")
+(def sanitize-metric-label-name-re #"[^a-zA-Z0-9_]+")
+
+
+
+(defn- good-name? [name] (s/and string? (re-matches metric-name-re name)))
+
+(s/def :metric/metric-type #{:counter :gauge :histogram :summary})
+(s/def :metric/metric-value double?)
```

Comment:
why `double?`?? what if you have an integer number?

Reply:
The underlying collection takes a double, but you are correct I should handle this later.